### PR TITLE
Modify to work with npm/import/require and webpack. 

### PR DESCRIPTION
**Pull request by dopry**
Allows require/import to work properly with angular-lock and removes the need to include both auth0-lock and angular-lock by setting up the package properly.  This will not be loadable directly through a browser script tag any more. 

* add auth0-lock as a package dependency.
* require Auth0Lock instead of depending on the global. 
* update docs to reflect usage with NPM.
* update version to 1.1.x to indicate a change. 

if you're using a bundler like webpack, you add my repo to your package.json to test. 

```
"dependencies": {
 "angular-lock": "https://github.com/dopry/angular-lock",
}
```

### CODE REVIEW

**Comment by chenkie** [comment]
Thanks for your PR :)

While I agree that this is the proper way to bring in the lib, we can't remove the ability to use it through a script tag since many people still rely on this. We need to be mindful of the fact that not everyone will be using a bundler.

That being said, I think this approach would make a good default if we can still support old fashioned script tags.

**Comment by dopry** [comment]
I scratched my itch. Will leave it out there for others... Very few Angular
developers use  script tags these days...  If I were you guys I'd write for
NPM/ require and bundle separately for script tags... But that's a
distribution issue not a code issue.  I ended up dropping all the Auth0
code altogether, and writing my own wrapper to the API...  It just doesn't
work with a contemporary ng1 project and the samples, modules, and
versioning are super confusing and hard to follow.

**Comment by dopry** [comment]
cleaning up my PR queue.
